tests/main/snapd-reexec/task.yaml

```diff
@@ -94,6 +96,13 @@ execute: |
     fi
     if [ "$IS_REEXEC" = true ]; then
         /usr/bin/env SNAPD_DEBUG=1 snap list 2>&1 | MATCH "DEBUG: restarting into \"$SNAPD_MOUNT_DIR/current/usr/bin/snap\""        
+
+        if [ "${SNAPD_DEB_FROM_REPO-}" = "false" ]; then
```

Comment:
I'm not familiar with this syntax. What does it do? I usually use `{foo:-}`.

Reply:
Almost the same:
> When not performing substring expansion, using the forms documented below (e.g., :‐), bash tests for a parameter that is unset or null.  Omitting the colon results in a test only for a parameter that is unset.

Since I only care of making this `-u` friendly, I only care about the unset case. But in this case they are both the same. If the variable is empty, `:-` will **replace** it with empty, and `-` will **keep** it empty.